src/System.Net.Sockets/tests/FunctionalTests/SendPacketsAsync.netcoreapp.cs

```diff
@@ -0,0 +1,218 @@
+            {
+                stream.Seek(s_testFileSize - 10, SeekOrigin.Begin);
+                SendPackets(type, new SendPacketsElement(stream, 0, 20), 20);
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    Assert.Equal(s_testFileSize - 10, stream.Position);
+
+                SendPackets(type, new SendPacketsElement(stream, 10, 20), 20);
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    Assert.Equal(s_testFileSize - 10, stream.Position);
+
+                SendPackets(type, new SendPacketsElement(stream, s_testFileSize - 20, 20), 20);
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    Assert.Equal(s_testFileSize - 10, stream.Position);
+            }
+        }
+
+        [Theory]
+        [InlineData(SocketImplementationType.APM)]
+        [InlineData(SocketImplementationType.Async)]
+        public void SendPacketsElement_FileStreamMultiPart_Success(SocketImplementationType type)
+        {
+            using (var stream = new FileStream(TestFileName, FileMode.Open, FileAccess.Read, FileShare.Read, 4096, useAsync: true))
+            {
+                var elements = new[]
+                {
+                    new SendPacketsElement(stream, 0, 20),
+                    new SendPacketsElement(stream, s_testFileSize - 10, 10),
+                    new SendPacketsElement(stream, 0, 10),
+                    new SendPacketsElement(stream, 10, 20),
+                    new SendPacketsElement(stream, 30, 10),
+                };
+                stream.Seek(s_testFileSize - 10, SeekOrigin.Begin);
+                SendPackets(type, elements, SocketError.Success, 70);
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    Assert.Equal(s_testFileSize - 10, stream.Position);
+
+                SendPackets(type, elements, SocketError.Success, 70);
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    Assert.Equal(s_testFileSize - 10, stream.Position);
+            }
+        }
+
+        [Theory]
+        [InlineData(SocketImplementationType.APM)]
+        [InlineData(SocketImplementationType.Async)]
+        public void SendPacketsElement_FileStreamAsyncMultiPart_Success(SocketImplementationType type)
+        {
+            using (var stream = new FileStream(TestFileName, FileMode.Open, FileAccess.Read, FileShare.Read, 4096, FileOptions.Asynchronous))
+            {
+                var elements = new[]
+                {
+                    new SendPacketsElement(stream, 0, 20),
+                    new SendPacketsElement(stream, s_testFileSize - 10, 10),
+                    new SendPacketsElement(stream, 0, 10),
+                    new SendPacketsElement(stream, 10, 20),
+                    new SendPacketsElement(stream, 30, 10),
```

Comment:
I might just be missing it, but we should have tests that validate a SendPacketsElement[] that contains buffers, paths, and streams, verifying they were all sent and all sent in the right order.  Do we?

Reply:
We don't, I agree that would be valuable. Working on it.

Reply:
Done...also checked that the bytes that are sent are actually the same, vs just the number sent.